lib/sshkit/utils.ex

```diff
@@ -2,4 +2,11 @@ defmodule SSHKit.Utils do
   def shellescape(value), do: value
 
   def shellquote(value), do: value
+
+  def strings_to_charlists(opts) when is_list(opts) do
+    Enum.map(opts, fn {k, v} -> {k, charlist(v)} end)    
```

Comment:
No elegant way springs to mind. What seems like it's going to make it unavoidably ugly is that we can't pattern match on the difference between a a charlist and a list of charlists, since they'll both pass an `is_list` guard.

One alternative strategy would be to take advantage of the fact that there are actually only a few valid values for these particular options.

Instead of taking a string or charlist, we could define some module attributes for the valid values. That doesn't prevent people from shooting themselves in the foot, but it does give them a convenient way not to.

The next (big) step beyond that would be to define Elixir data structures for each of Erlang ssh options, then we could enforce a whole bunch of stuff. But that's a lot of code for what are arguably edge cases.